library/src/scala/Tuple.scala

```diff
@@ -167,25 +214,38 @@ object Tuple {
    *  ```
    *  @syntax markdown
    */
-  type Filter[Tup <: Tuple, P[_] <: Boolean] <: Tuple = Tup match {
+  type Filter[X <: Tuple, P[_] <: Boolean] <: Tuple = X match
     case EmptyTuple => EmptyTuple
-    case h *: t => P[h] match {
+    case h *: t => P[h] match
       case true => h *: Filter[t, P]
       case false => Filter[t, P]
-    }
-  }
 
-  /** Given two tuples, `A1 *: ... *: An * At` and `B1 *: ... *: Bn *: Bt`
-   *  where at least one of `At` or `Bt` is `EmptyTuple` or `Tuple`,
-   *  returns the tuple type `(A1, B1) *: ... *: (An, Bn) *: Ct`
-   *  where `Ct` is `EmptyTuple` if `At` or `Bt` is `EmptyTuple`, otherwise `Ct` is `Tuple`.
+  /** A tuple consisting of those indices `N` of tuple `X` where the predicate `P`
+   *  is true for `Elem[X, N]`. Indices are type level values <: Int.
+   */
+  type IndicesWhere[X <: Tuple, P[_] <: Boolean] =
```

Comment:
It could be upper bounded by Tuple

Reply:
I meant:
```suggestion
  type IndicesWhere[X <: Tuple, P[_] <: Boolean] <: Tuple =
```

Reply:
No that does not work since the right hand side is not a match type.

Reply:
Indeed, my mistake